### DISCUSSION TITLE
How can I change the location where the blog tag pages are displayed?

### DISCUSSION THREAD

**Discussion by JetGum** [datenstrom/community#839] [Ask a question] [answered]
I am trying to find a solution for displaying the tag pages of my blog in a different location than intended by the system.

Let me explain:

What I would love to do is to to change the configuration so that my tag pages can be displayed not in root but in a folder, for example, /tag/. 

Unfortunately, the tag pages use blog-start as a layout file, which in my case allows only blog pages with the "home" tag to display.
(Background: I have a blog with BlogStartLocation: / (= root). So the homepage of the blog (= page.md in root) has the layout:blog-start and BlogWithFilter:1, which means that I only show blog entries with the tag "home" on homepage. Not all my blog pages have this tag because I do not want all blog pages to appear on the start page.)

This is the "tagcloud"-code that generates the tags in a sidebare of a blog page:

`<div class="tagcloud"><?php echo $this->yellow->page->getExtra("links") ?><?php if ($this->yellow->page->isExisting("tag")): ?><h3>Related topics</h3><?php $tagCounter = 0; foreach (preg_split("/\s*,\s*/", $this->yellow->page->get("tag")) as $tag) { if (++$tagCounter>1) echo " "; echo "<a class=\"btn\" href=\"".$this->yellow->page->getPage("blogStart")->getLocation(true).$this->yellow->toolbox->normaliseArguments("tag:$tag")."\">".htmlspecialchars($tag)."</a>"; } ?></div>`

I have succeeded in changing the URL where the tag pages are displayed by adding "normaliseArguments("tag/tag:$tag")", but this results in a 404 for root/tag/tag:$tag URL.

Any ideas how I as a php-noob could change this?

Any help as always greatly appreciated.

Cheers,
Carsten

**Comment by JetGum** [comment]
Case solved.

Instead of trying to change the layout file for tag result page, I pointed the tags toward a search query.

So, in my sidebar, I have "Latest posts" (just a [blogchanges / 0] shortcode in a *.md-file.
Plus "Related topics", which are linked to a search: (search:/tag:$tag). 

Works fine for me, and here's the code.

```
<div class="tagcloud">
<h3>Latest Posts</h3>
<?php echo $this->yellow->page->getPage("latest-posts")->getContent() ?>
<br>
<?php echo $this->yellow->page->getExtra("links") ?>
<?php if ($this->yellow->page->isExisting("tag")): ?>
<h3>Related topics</h3>
<?php $tagCounter = 0; foreach (preg_split("/\s*,\s*/", $this->yellow->page->get("tag")) as $tag) { if (++$tagCounter>1) echo " "; echo "<a class=\"btn\" href=\"".$this->yellow->page->getBase(true).$this->yellow->toolbox->normaliseArguments("/search/tag:$tag")."\">".htmlspecialchars($tag)."</a>"; } ?>
<?php endif ?>
</div>
```

PS: There's only one odd thing I noticed on the search result page. The title "**Search: Recent changes**" now appears as h1. Anybody knows why?

Cheers, 

Carsten

**Reply by JetGum** [reply]
Found the setting... 
In yellow-language.ini, there's a setting "SearchSpecialChanges: Recent changes".
This can be changed into SearchSpecialChanges: Your search" or anything else that fits.
Cheers,
Carsten

**Comment by schulle4u** [comment, marked as answer]
You can customize the layout of search results by editing the file `system/layouts/search.html`, or create a theme-specific layout file, e.g. `system/layouts/berlin-search.html`.

**Comment by annaesvensson** [comment]
As Steffen wrote this is something you can customise in your own layout files, though it requires some basic programming knowledge. The API for developers should have everything someone needs to access pages, filter pages and handle arguments. Take a look at an extension's source code, it uses the same API, it can give you some examples of how to use it in your own layout files.

**Reply by JetGum** [reply]
Dear Anna,

thanks for your encouraging words! I am still in the "look at extension code and copy, paste, try and error" phase with PHP. Once this gets too time-consuming and as soon as I come to terms with the basics, I would love to adjust the system and use the API.

Cheers, Carsten

PS: Any recommendations as to what resources would help me while looking at an extension's source code?

**Reply by annaesvensson** [reply]
Do you mean resources for learning PHP programming?

**Reply by JetGum** [reply]
Yes, there are a myriad of resources out there. In particular the syntax and "punctuation" of php in html is not quite self-evident, I believe. thanks!

**Reply by annaesvensson** [reply]
Maybe just maybe, start a new discussion and you get tips from our community. 😄
